R/assemble.R

```diff
@@ -241,3 +252,55 @@ de_duplicate_names <- function(x) {
   )
   x
 }
+
+
+# does a data frame contain duplicate rows
+# @param x a data.frame
+# @return logical, TRUE if there are any duplicates in the data.frame
+contains_duplicates = function(x) {
+  sum(find_duplicates(x)) > 0
+}
+
+# @param x a data.frame
+# @return a logical vector of the duplicated rows
+find_duplicates = function(x) {
+  duplicated(dplyr::select_if(x, is.atomic))
+}
+
+# @param x a data.frame
+duplicate_drop = function(x) {
+  ix_dupes = find_duplicates(x)
+  if (sum(ix_dupes) > 0) {
+    warning(
+      "There are duplicate responses, duplicates are dropped in
+      the results. Set fix_duplicates = 'keep' to retain them."
+    )
+  }
+  x[!ix_dupes, ]
+}
+
+# @param x a data.frame
+duplicate_keep = function(x) {
+  if (contains_duplicates(x)) {
+    warning(
+      "There are duplicate responses, duplicates are retained in
+      the results. Set fix_duplicates = 'drop' to remove them."
+    )
+  }
+  x
+}
+
+# Deal with duplicate data by throwing an error
+duplicate_error = function(x) {
+  # There should not be duplicate rows here, but putting this here in case of oddities like #27
+  assertthat::assert_that(!contains_duplicates(x),
+    msg = paste0(
+      "There are duplicated rows in the responses, maybe a situation like #27\n",
```

Comment:
```suggestion
      "There are duplicated rows in the responses.\n",
```
Hoping this PR can close #27 and #74 for now. It seems that when thing start to leak through the API, there are new issues with duplicate rows, and likely aren't related to #27 anymore.